asmcomp/amd64/selection.ml

```diff
@@ -238,6 +238,14 @@ method! select_operation op args dbg =
           (Ispecific Isextend32, [k])
         | _ -> super#select_operation op args dbg
       end
+  (* Recognize zero extension *)
+  | Cand ->
+    begin match args with
+    | [arg; Cconst_int (0xffff_ffff, _)]
```

Comment:
(I understand you are recognizing patterns produced
earlier in the pipeline but, to accommodate user-written
code, it might be interesting to also match
`[Cconst_int (0xffff_ffff, _); arg]`.)

Reply:
Good idea, thanks!